File: lib/HelloWorld.cpp
#include <HelloWorld.hpp>

HelloWorld::HelloWorld() {
  std::cout << "HelloWorld: Constructor" << std::endl;
}

void HelloWorld::printHello() { std::cout << prefix << "Hello"; }

void HelloWorld::print() { std::cout << prefix << "Hello World!" << std::endl; }

void HelloWorld::print(std::string something) {
  std::cout << prefix << something << std::endl;
}

File: example/HelloWorld.py
#!/usr/bin/env python3

from ViennaExample import HelloWorld

# test construction
hw = HelloWorld()

# print "Hello, World!"
hw.print()

# print our own text
hw.print("This is an example!")

# set new prefix for output
hw.prefix = "Output: "

# test output with prefix
hw.print()
